presto-main/src/main/java/com/facebook/presto/operator/unnest/Unnester.java

```diff
@@ -52,65 +51,59 @@ public final void resetInput(Block block)
         for (int i = 0; i < getChannelCount(); i++) {
             unnestBlockBuilders[i].resetInputBlock(getElementsBlock(i));
         }
-        currentPosition = 0;
     }
 
-    public final int getCurrentUnnestedLength()
+    public abstract int getChannelCount();
```

Comment:
- getChannelCount() method is not needed; the result can be computed as unnestBlockBuilders.length

```
    private final int channelCount;

    protected Unnester(UnnestBlockBuilder... unnestBlockBuilders)
    {
        this.unnestBlockBuilders = requireNonNull(unnestBlockBuilders, "unnestBlockBuilders is null");
        this.channelCount = unnestBlockBuilders.length;
    }
```

Reply:
It's still needed by the operator. But I have changed the implementation of it to
```
public int getChannelCount()
    {
        return channelCount;
    }
```